sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileSystems.java

```diff
@@ -99,44 +179,177 @@ public static WritableByteChannel create(ResourceId resourceId, CreateOptions cr
    * @param resourceId the reference of the file-like resource to open
    */
   public static ReadableByteChannel open(ResourceId resourceId) throws IOException {
-    return getFileSystemInternal(resourceId).open(resourceId);
+    return getFileSystemInternal(resourceId.getScheme()).open(resourceId);
   }
 
-  /********************************** METHODS FOR REGISTRATION **********************************/
+  /**
+   * Copies a {@link List} of file-like resources from one location to another.
+   *
+   * <p>The number of source resources must equal the number of destination resources.
+   * Destination resources will be created recursively.
+   *
+   * <p>{@code srcResourceIds} and {@code destResourceIds} must have the same scheme.
+   *
+   * <p>It doesn't support copying globs.
+   *
+   * @param srcResourceIds the references of the source resources
+   * @param destResourceIds the references of the destination resources
+   */
+  public static void copy(
+      List<ResourceId> srcResourceIds,
+      List<ResourceId> destResourceIds,
+      MoveOptions... moveOptions) throws IOException {
```

Comment:
I was following the nio interface, which uses CopyOption in both copy() and move(). 
https://docs.oracle.com/javase/7/docs/api/java/nio/file/CopyOption.html
(And, I also used the same options in delete()).

I think CopyOptions and RenameOptions could share many configuration options.
If we need copy specific options, we can make CopyOptions to extend MoveOptions. This interface accept both.